### ISSUE TITLE
sway crash, maybe related to copy/paste

### ISSUE THREAD

**Issue by Soundtoxin** [swaywm/sway#4726]
* Sway Version:
sway version 1.2

* Debug Log & Configuration File:
[config and debug log](https://gist.github.com/Soundtoxin/f569b7661aa0109b5535cd2e84de3656)

* Finally, explain the steps you took in plain English to reproduce the problem below.
I had one crash trying to copy a filename in Deluge, another (this one I recorded) I was trying to middle-click paste into Firefox to paste.debian.org to show someone some dmesg lines.
Here those are in case they're helpful.
```
[19648.086755] .sway-wrapped[1973]: segfault at 4b000000768 ip 00007fafe0da7df9 sp 00007ffc46f44f20 error 4 in libwlroots.so.3.6.1[7fafe0d9f000+4f000]
[19648.086764] Code: 00 00 41 8b 4f 04 48 8d 70 e8 48 39 c2 75 1b e9 75 07 00 00 0f 1f 44 00 00 48 8b 46 20 48 8d 70 e8 48 39 c2 0f 84 5f 07 00 00 <3b> 48 e8 75 ea 41 8b 57 08 48 89 df e8 76 f3 ff ff e9 49 07 00 00
[38184.698656] traps: gdm[21585] trap int3 ip:7f8dbaba2c75 sp:7ffcb33bfc30 error:0 in libglib-2.0.so.0.6200.1[7f8dbab69000+80000]
[38185.513770] traps: gdm[21615] trap int3 ip:7f250e104c75 sp:7fff4a6775b0 error:0 in libglib-2.0.so.0.6200.1[7f250e0cb000+80000]
[38186.266535] traps: gdm[21639] trap int3 ip:7feb10ac4c75 sp:7ffd60230160 error:0 in libglib-2.0.so.0.6200.1[7feb10a8b000+80000]
[85411.921055] .sway-wrapped[6241]: segfault at ffffffffffffffe8 ip 00007f44fdeac620 sp 00007ffd6b6f6180 error 5 in libwlroots.so.3.6.1[7f44fdea3000+4f000]
[85411.921064] Code: 00 00 48 8b 83 38 03 00 00 48 8d 93 30 03 00 00 45 8b 47 04 4c 8d 68 e8 48 39 c2 74 24 44 39 40 e8 75 11 e9 28 01 00 00 66 90 <44> 3b 40 e8 0f 84 1c 01 00 00 49 8b 45 20 4c 8d 68 e8 48 39 c2 75
[85412.644615] [drm:intel_set_cpu_fifo_underrun_reporting [i915]] *ERROR* uncleared fifo underrun on pipe B
[85412.644632] [drm:intel_cpu_fifo_underrun_irq_handler [i915]] *ERROR* CPU pipe B FIFO underrun
```

* Stack Trace, if sway crashes:
I'm on NixOS with systemd but the suggested command to get the coredump did not seem to work. I get "No match found."

**Comment by emersion** [comment]
```
2019-11-13 09:36:56 - [wayland] dup failed: Too many open files
```

**Comment by Soundtoxin** [comment]
I never ran into this issue on Guix System, and I'm using this machine roughly the same way. What would lead to a bunch of files being open in some sort of invisible way?

**Comment by martinetd** [comment]
It's not necessarily invisible, if this isn't a one-time crash (e.g. whatever behaviour caused this is likely to happen again) could you post the content of something like `lsof -p $(pgrep -x sway)` to see what kind of open files sway has?

As a workaround you might want to increase the 'open files' ulimit setting as well, but you likely have some sort of fd leak we need to identify and fix. If you can do the lsof command a few times (or just count the number of fds in /proc/<pid>/fd) and see if you can do something that reliably increases the number then it might help fix the issue.

**Comment by Soundtoxin** [comment]
> could you post the content of something like `lsof -p $(pgrep -x sway)` to see what kind of open files sway has?

That command didn't work for me, seemingly because the pgrep portion on its own wasn't picking up any sway processes.

```
brad@anpan:~/ > lsof -p $(pgrep -x sway)                                       
lsof: no process ID specified
lsof 4.93.2                            
 latest revision: https://github.com/lsof-org/lsof
 latest FAQ: https://github.com/lsof-org/lsof/blob/master/00FAQ
 latest (non-formatted) man page: https://github.com/lsof-org/lsof/blob/master/Lsof.8
 usage: [-?abhKlnNoOPRtUvVX] [+|-c c] [+|-d s] [+D D] [+|-E] [+|-e s] [+|-f[gG]]
 [-F [f]] [-g [s]] [-i [i]] [+|-L [l]] [+m [m]] [+|-M] [-o [o]] [-p s]
 [+|-r [t]] [-s [p:s]] [-S [t]] [-T [t]] [-u s] [+|-w] [-x [fl]] [--] [names]
Use the ``-h'' option to get more help information.
```
pgrep without the -x arg finds a few
```
brad@anpan:~/ > pgrep sway                                                     
22355                                                                          
22359                                                                          
22361                                                                          
```

and here is htop using f4 (filter) and typing "sway". Perhaps the unusal paths on NixOS are a problem.
```
22355 brad       20   0  341M  170M  152M S  5.2  4.5  4:19.47 /nix/store/kncf2qj9bhjn42wn7pri1a96ya24gd4r-sway-1.2/bin/sway                                   
22361 brad       20   0 22916 14016 11684 S  2.0  0.4  0:12.57 swaybar -b bar-0
22359 brad       20   0 46108 37784 25688 S  0.0  1.0  0:00.07 swaybg -o * -i /home/brad/pictures/wallpaper/1529004448340.jpg -m fill
22357 brad       20   0  341M  170M  152M S  0.0  4.5  0:00.00 /nix/store/kncf2qj9bhjn42wn7pri1a96ya24gd4r-sway-1.2/bin/sway
13156 brad       20   0  126M  3340  2940 S  0.0  0.1  0:00.00 /bin/sh -c wofi --show run | xargs swaymsg exec
13158 brad       20   0  125M  2100  1928 S  0.0  0.1  0:00.00 xargs swaymsg exec
15888 brad       20   0  126M  3304  2904 S  0.0  0.1  0:00.00 /bin/sh -c wofi --show run | xargs swaymsg exec
15890 brad       20   0  125M   204   164 S  0.0  0.0  0:00.00 xargs swaymsg exec
```

> (or just count the number of fds in /proc//fd) and see if you can do something that reliably increases the number then it might help fix the issue.

It seems I don't have /proc/fd! I don't know if it's more weird NixOS stuff or if maybe you meant the similarly-named /proc/fb. Here is my proc  directory:
```
brad@anpan:~/ > ls /proc/fd                                                    
ls: cannot access '/proc/fd': No such file or directory         
brad@anpan:~/ > ls /proc                                                       
1/      1348/   181/    20292/  2495/   40/    8839/        kpagecgroup
10/     13554/  182/    20293/  25/     43/    8840/        kpagecount
100/    13591/  183/    20352/  2507/   44/    9/           kpageflags
101/    13685/  18325/  20354/  2542/   45/    9001/        loadavg
10325/  14/     18361/  20355/  2600/   47/    93/          locks        
106/    1447/   184/    20356/  27163/  48/    94/          meminfo 
1085/   14611/  1841/   2060/   27333/  489/   95/          misc       
10966/  1463/   18461/  20746/  27438/  49/    98/          modules   
11/     1470/   18474/  20965/  28/     50/    9831/        mounts
1131/   14785/  185/    20988/  2806/   505/   99/          mtd
1156/   15/     18516/  20989/  2831/   5146/  acpi/        mtrr
1157/   1525/   1853/   21595/  2855/   515/   asound/      net
1158/   15294/  1854/   21619/  28631/  53/    buddyinfo    pagetypeinfo
1159/   15308/  1859/   21643/  28748/  54/    bus/         partitions
11667/  15484/  186/    22355/  2876/   55/    cgroups      pressure/
11769/  15888/  1860/   22359/  28957/  56/    cmdline      sched_debug
12/     15889/  187/    22361/  2897/   57/    config.gz    scsi/
1214/   15890/  188/    22363/  28973/  5773/  consoles     self
1220/   16009/  189/    22367/  29/     58/    cpuinfo      slabinfo
1228/   16015/  19/     22373/  29177/  589/   crypto       softirqs
1280/   16062/  190/    22375/  2923/   591/   devices      stat
13/     1628/   1969/   22377/  2947/   592/   diskstats    swaps
1304/   17009/  19805/  22378/  2968/   594/   dma          sys/
13097/  17142/  19879/  22381/  2994/   601/   driver/      sysrq-trigger
13156/  17145/  19886/  22393/  3/      615/   execdomains  sysvipc/
13157/  17176/  19970/  22394/  30/     628/   fb           thread-self
13158/  17183/  2/      22524/  3014/   629/   filesystems  timer_list
13227/  17214/  20/     226/    3037/   645/   fs/          tty/
13267/  17360/  200/    227/    3157/   685/   interrupts   uptime
1327/   17361/  20214/  22785/  33/     693/   iomem        version
13272/  17370/  20256/  23/     3344/   7031/  ioports      vmallocinfo
13280/  17381/  20257/  2386/   34/     791/   irq/         vmstat
1331/   178/    20258/  24/     35/     7991/  kallsyms     zoneinfo
13373/  179/    20259/  2413/   38/     8/     kcore
1340/   18/     20263/  2460/   3847/   8185/  keys
1341/   180/    20274/  2461/   39/     8798/  key-users
1347/   18039/  20275/  2486/   4/      8799/  kmsg
```

**Comment by martinetd** [comment]
Err, sorry for both - the former is indeed the full path being the problem or rather me being too optimistic, just type in sway's main pid in - 22355 here.

The later is weirder, github apparently eats words between angle brackets? I typed in [bracket]pid[bracket] in between /proc/ and /fd/... Anyway, please have a look at `/proc/22355/fd` in this case again. It really is the same as what lsof outputs, but might be easier to look at the directory directly when trying to make the number grow, it really shouldn't grow when using sway normally (it will grow when new wayland clients appear for example, but should get back to the former number when they get closed)

**Comment by Soundtoxin** [comment]
> Anyway, please have a look at `/proc/22355/fd` in this case again.

Okay, hope I'm doing this right. I get different numbers with these two approaches, but it looks like the lsof command has a lot of different types of stuff.
```
brad@anpan:~/ > lsof -p 22355 | wc -l
386
brad@anpan:~/ > ls /proc/22355/fd | wc -l
139
```

**Comment by martinetd** [comment]
I wasn't clear again, can you come in #sway on irc to iterate instead of spamming 252 people every message?

(the lsof command's output is interesting in itself, it has more informations than the /proc/x/fd because it tells what kind of fd are open and we might understand something from what we see here ;
on the other hand the number of fd itself is something interesting to track over time if you can get it to increase reliably, because that would let you identify some specific action in what you were doing that causes the increase. 139 in itself is pretty normal I would say, not something that would cause the log messages you had in your first post)

**Comment by Soundtoxin** [comment]
Okay, after some discussion in irc, I found that copying text a few times increased my fd value to 1023, Just a hair below the max. It seems `wl-paste --watch` and `clipman` cause known issues and I had those executing in my sway config. I have now removed them. Copying a bunch of text in a row no longer seems to increase the value. This issue can be marked as a duplicate.

**Comment by bugaevc** [comment]
> It seems `wl-paste --watch` and `clipman` cause known issues and I had those executing in my sway config. I have now removed them. Copying a bunch of text in a row no longer seems to increase the value.

Could you please check if it's actually `wl-paste --watch` or `clipman` causing the problem? Try running e.g. `wl-paste --watch echo new offer`, and if that's enough to reproduce it, report it over at wl-clipvoard repo.
